internal/constraints/cache.go

```diff
@@ -114,7 +114,7 @@ func (c *Cache) prepareEnvironment(
 ) (*cel.Env, error) {
 	env, err := env.Extend(
 		cel.Types(rules.Interface()),
-		cel.Variable("this", c.getCELType(fieldDesc, forItems)),
+		cel.Variable("this", expression.ProtoFieldToCELType(fieldDesc, true, forItems)),
```

Comment:
For my own knowledge, why is generic set to true here?

Reply:
Here we are compiling the standard constraint expressions the first time we see them, and we use a concrete field it's attached to as the basis for type information. `repeated` and `map` fields are generic over their elements, so when we build (say) the `unique` expression for `RepeatedRules`, we cannot assume that the element type of the list is always going to be that of the field we're using since another repeated field of a different type might also specify unique.